osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/configuration/service/GetConfigurationObjectServiceDsmr4.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2017 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.configuration.service;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.openmuc.jdlms.GetResult;
+import org.openmuc.jdlms.datatypes.BitString;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagTypeDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationFlagsDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ConfigurationObjectDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GprsOperationModeTypeDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GetConfigurationObjectServiceDsmr4 extends GetConfigurationObjectService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GetConfigurationObjectServiceDsmr4.class);
+
+    private static final int NUMBER_OF_CONFIGURATION_OBJECT_ELEMENTS = 2;
+    private static final int INDEX_OF_GPRS_OPERATION_MODE = 0;
+    private static final int INDEX_OF_CONFIGURATION_FLAGS = 1;
+
+    private final DlmsHelper dlmsHelper;
+
+    public GetConfigurationObjectServiceDsmr4(final DlmsHelper dlmsHelper) {
```

Comment:
I think you need to @Autowired the constructor, otherwise at runtime the DlmsHelper will be null.

Reply:
No, Spring Boot does this for us automatically (because DlmsHelper is a Spring bean)

Reply:
ok, I did not know this. with regular spring you need to do this. but Spring Boot is different ?

Reply:
Indeed.